waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm

```diff
@@ -0,0 +1,107 @@
+// QUALITY COPYPASTA
+/turf/closed/indestructable/bluespace_cascade
+	name = "Bluespace Cascade Foam"
+	desc = "The unimaginable consequences of tampering with Bluespace. The Universe will never be the same."
+	icon='waspstation/icons/turfs/walls/bs_cascade.dmi'
+	icon_state = "bluespacecrystal1"
+	light_range = 5
+	light_power = 2
+	light_color="#0066FF"
+	plane = CAMERA_STATIC_PLANE
+
+	var/next_check=0
+	var/list/avail_dirs = list(NORTH,SOUTH,EAST,WEST)
+	var/spread_process_timer
+
+/turf/closed/indestructable/bluespace_cascade/New()
+	spread_process_timer = addtimer(CALLBACK(src, .process), 30, TIMER_OVERRIDE | TIMER_STOPPABLE)
+	return ..()
+
+/turf/closed/indestructable/bluespace_cascade/Destroy()
+	deltimer(spread_process_timer)
+	return ..()
+
+/turf/closed/indestructable/bluespace_cascade/process()
+	// No more available directions? Shut down process().
+	if(!length(avail_dirs))
+		deltimer(spread_process_timer)
+
+	// Choose a direction.
+	var/pdir = pick(avail_dirs)
+	avail_dirs -= pdir
+	var/turf/T=get_step(src,pdir)
+	if(istype(T, /turf/closed/indestructable/bluespace_cascade))
+		avail_dirs -= pdir
+		spread_process_timer = addtimer(CALLBACK(src, .process), 30, TIMER_OVERRIDE | TIMER_STOPPABLE)
+
+	// EXPAND DONG
+	if(isturf(T))
+		// Nom.
+		T.singularity_act()
+		T.ChangeTurf(/turf/closed/indestructable/bluespace_cascade, flags= CHANGETURF_INHERIT_AIR)
+		for(var/atom/A in T.contents)
+			A.Destroy()
+		var/turf/closed/indestructable/bluespace_cascade/BC = T
+		BC.New()
+
+/turf/closed/indestructable/bluespace_cascade/attack_paw(mob/user as mob)
+	return attack_hand(user)
+
+/turf/closed/indestructable/bluespace_cascade/attack_robot(mob/user as mob)
+	if(Adjacent(user))
+		return attack_hand(user)
+	else
+		to_chat(user, "<span class = \"warning\">What the fuck are you doing?</span>")
```

Comment:
is there a reason this is here but not on normal attack_hand?

Reply:
copied from the /vg/ implementation